kernelci/cli/docker.py

```diff
@@ -0,0 +1,203 @@
+class cmd_build(DockerBuildGenerateCommand):  # pylint: disable=invalid-name
+    """Build image"""
+
+    opt_args = DockerCommand.opt_args + [
+        {
+            'name': '--build-arg',
+            'action': 'append',
+            'help': "Build argument to pass to docker build",
+        },
+        {
+            'name': '--no-cache',
+            'action': 'store_true',
+            'help': "Disable Docker cache when building the image",
+        },
+        {
+            'name': '--push',
+            'action': 'store_true',
+            'help': "Push the image to Docker Hub after building",
+        },
+        {
+            'name': '--verbose',
+            'action': 'store_true',
+            'help': "Verbose output",
+        },
+    ]
+
+    @classmethod
+    def _build_image(cls, dockerfile, tag, buildargs, nocache):
+        client = docker.from_env()
+        dockerfile_obj = io.BytesIO(dockerfile.encode())
+        return client.images.build(
+            fileobj=dockerfile_obj, tag=tag, buildargs=buildargs,
+            nocache=nocache,
+        )
+
+    @classmethod
+    def _dump_dockerfile(cls, dockerfile):
+        sep = '--------------------------------------------------' * 2
+        print()
+        print(sep)
+        print(dockerfile)
+        print(sep)
+        print()
+
+    @classmethod
+    def _dump_log(cls, build_log):
+        for chunk in build_log:
+            stream = chunk.get('stream') or ""
+            for line in stream.splitlines():
+                print(line)
+
+    @classmethod
+    def _push_image(cls, base_name, tag_name):
+        client = docker.from_env()
+        push_log_json = client.images.push(base_name, tag_name)
+        return list(
+            json.loads(json_line)
+            for json_line in push_log_json.splitlines()
+        )
+
+    @classmethod
+    def _dump_push_log(cls, log):
+        for line in log:
+            if 'status' in line and 'progressDetail' not in line:
+                print(line['status'])
+
+    def __call__(self, configs, args):
+        base_name, tag_name, name = self._gen_image_name(args)
+
+        dockerfile = self._get_dockerfile(args)
+        if args.verbose:
+            self._dump_dockerfile(dockerfile)
+
+        print(f"Building {name}")
+        buildargs = dict(
+            barg.split('=') for barg in args.build_arg
+        ) if args.build_arg else {}
+        _, build_log = self._build_image(dockerfile, name, buildargs,
+                                         args.no_cache)
+        if args.verbose:
+            self._dump_log(build_log)
+        if args.push:
+            print(f"Pushing {name} to Docker Hub")
+            push_log = self._push_image(base_name, tag_name)
+            if args.verbose:
+                self._dump_push_log(push_log)
+            for line in push_log:
+                error = line.get('errorDetail')
+                if error:
+                    print(error['message'])
+                    sys.exit(1)
```

Comment:
```python
return False
```
is the regular way to report errors with the `kci` command line tools.

Reply:
Fixed it.